torchvision/datasets/mnist.py

```diff
@@ -36,15 +36,20 @@ class MNIST(data.Dataset):
     training_file = 'training.pt'
     test_file = 'test.pt'
 
-    def __init__(self, root, train=True, transform=None, target_transform=None, download=False):
+    def __init__(self, root, train=True, transform=None, target_transform=None,
+                 download=False, from_local=False):
```

Comment:
I think there is no need to add an extra argument `from_local`, but instead separate `download` in two functions: `download` and `process`, and assert that if the original file is already downloaded but the preprocessed file doesn't exist you pre-process it.